openhcl/underhill_core/src/nvme_manager.rs

```diff
@@ -232,11 +236,15 @@ impl NvmeManagerWorker {
         let driver = match self.devices.entry(pci_id.to_owned()) {
             hash_map::Entry::Occupied(entry) => entry.into_mut(),
             hash_map::Entry::Vacant(entry) => {
-                let device =
-                    VfioDevice::new(&self.driver_source, entry.key(), self.dma_buffer.clone())
-                        .instrument(tracing::info_span!("vfio_device_open", pci_id))
-                        .await
-                        .map_err(InnerError::Vfio)?;
+                let device = VfioDevice::new(
+                    &self.driver_source,
+                    entry.key(),
+                    (self.dma_buffer_spawner)(format!("nvme_{}", entry.key()))
```

Comment:
Slightly worried about the stability of this key. But I think it's OK for now.

Reply:
(We need to do some experiments to see what happens if two VPCI buses have colliding PCI IDs. I don't think we handle this well elsewhere, already.)

Reply:
Yeah I'm open to other ideas here... And I think that if the key collides, we already have problems because that means we'd only have one driver instance in nvme_manager.

Reply:
Right